### PR TITLE
Handle Selection Not Collapsed for delimiters

### PR DESCRIPTION
**Pull request by BryanValverdeU**
Handle a scenario where the delimiter is removed when user selects the text + delimiter.

Repro
1. Add Entity
2. type Next and Before the Entity
3. Select the text after/before the entity  
4. press a key the modified the content

Actual:
The delimiter is removed in some scenarios

Expected:
Do not remove the delimiter

To fix modify the selection and prevent the user typing inside of the delimiter when selection is not collapsed

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts [hunk 1]
```diff
@@ -111,57 +112,97 @@ export function inlineEntityOnPluginEvent(event: PluginEvent, editor: IEditor) {
...
 
-                        editor.runAsync(asyncEditor => {
-                            if (isAfter) {
-                                const elAfter = asyncEditor.getElementAtCursor();
-                                removeDelimiterAttr(elAfter);
-                            }
-                            if (element) {
-                                removeNode(element);
-                            }
+                        editor.runAsync(aEditor => {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts:115]: this function is already too long (~140lines) let's break it into several pieces.

**Comment by JiuqingSong** [comment]
> Insert 2 entities, then type some text, press Left, need to press Left twice to move to between entites

Here's the snapshot after I insert 2 entities, I see one delimiter is missing between them

```html
<span class="entityDelimiterBefore">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa_1 _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><br><!--{"type":0,"isDarkMode":false,"start":[4,0,1],"end":[4,0,1]}-->
```

**Comment by BryanValverdeU** [comment]
> > Insert 2 entities, then type some text, press Left, need to press Left twice to move to between entites
> 
> Here's the snapshot after I insert 2 entities, I see one delimiter is missing between them
> 
> ```
> <span class="entityDelimiterBefore">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa_1 _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><br>
> ```

This was introduced in last PR.  Added a bug.

**Comment by JiuqingSong** [comment]
> > > Insert 2 entities, then type some text, press Left, need to press Left twice to move to between entites
> > 
> > 
> > Here's the snapshot after I insert 2 entities, I see one delimiter is missing between them
> > ```
> > <span class="entityDelimiterBefore">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><span class="_Entity _EType_aaaaaaaaaaa _EId_aaaaaaaaaaa_1 _EReadonly_1" contenteditable="false" style="display: inline-block;"><span data-hydrated-html=""><b>bbbbbbbb</b></span></span><span class="entityDelimiterAfter">​</span><br>
> > ```
> 
> This was introduced in last PR. Added a bug.

ok

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**